### PR TITLE
Approve worklow button

### LINKED ISSUE #13: Webhook: add button to approve build
Add a clickable button redirecting to the CircleCI panel to approve the build.

- [Mattermost buttons docs](https://docs.mattermost.com/developer/interactive-messages.html#message-buttons)

image

### PR DESCRIPTION
**Pull request by nathanaelhoun**
## Changes

- [x] Approve workflow via a button in the notification
- [x] Update README.md

Closes #13

## Changelog
```
## New features
- Approve held workflows from the notification in Mattermost
```

### CODE REVIEW

**Comment by nathanaelhoun** [comment]
@darkLord19 This is WIP work, Pradeep and I encounter some problem with the button feature.

File: server/plugin/approve.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	}
+
+	responsePost := &model.Post{
+		ChannelId: requestData.ChannelId,
+		RootId:    requestData.PostId,
+		UserId:    p.botUserID,
+	}
+
+	workFlowID := fmt.Sprintf("%v", requestData.Context["WorkflowID"])
+	jobs, err := circle.GetWorkflowJobs(circleciToken, fmt.Sprintf("%v", requestData.Context["WorkflowID"]))
```
Comment (nathanaelhoun) [inline server/plugin/approve.go]: Use the previously computed value
```suggestion
	jobs, err := circle.GetWorkflowJobs(circleciToken, workFlowID)
```

**Review by nathanaelhoun — COMMENTED** [review]
A little change, the rest is good ! (I can't approve because I have open this PR. Feel free to approve it yourself/open a new one)

File: README.md [hunk 2]
```diff
@@ -166,6 +167,14 @@ Subscribe a channel to notifications from a CircleCI project.
     -   You may add it to the orb as a parameter, but this is discouraged as it should be treated like a secret
     -   You should add it as a Environment Variable named `MM_WEBHOOK`, through the [CircleCI UI](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-project) or using the plugin: `/circleci project env add MM_WEBHOOK <webhook-url>`
 
+### Approve a workflow from Mattermost
+
+You receive add notifications for [holded workflows](https://circleci.com/docs/2.0/workflows/#holding-a-workflow-for-a-manual-approval) and approved them directly from your Mattermost channel.
```
Comment (darkLord19) [inline README.md]:
```suggestion
You can receive add notifications for [held workflows](https://circleci.com/docs/2.0/workflows/#holding-a-workflow-for-a-manual-approval) and approve them directly from your Mattermost channel.
```

File: README.md [hunk 3]
```diff
@@ -166,6 +167,14 @@ Subscribe a channel to notifications from a CircleCI project.
...
+
+Learn how to set up a holded workflow notification in [our Orb documentation](https://circleci.com/developer/orbs/orb/nathanaelhoun/mattermost-plugin-notify#jobs-approval-notification).
```
Comment (darkLord19) [inline README.md]:
```suggestion
Learn how to set up a held workflow notification in [our Orb documentation](https://circleci.com/developer/orbs/orb/nathanaelhoun/mattermost-plugin-notify#jobs-approval-notification).
```

**Review by darkLord19 — APPROVED** [review]

**Pull request merged**